iModelCore/ecobjects/src/ECSchema.cpp

```diff
@@ -3211,6 +3211,18 @@ ECSchemaPtr SearchPathSchemaFileLocater::_LocateSchema(SchemaKeyR key, SchemaMat
     return schemaOut;
     }
 
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+ECSchemaPtr StringSchemaLocater::_LocateSchema(SchemaKeyR key, SchemaMatchType matchType, ECSchemaReadContextR schemaContext)
+    {
+    ECSchemaPtr schema;
+    auto schemaXml = m_schemaXml.find(key.m_schemaName);

```

Comment:
This is hacky, it works for tests, but for published code we should improve this. You have to check the schema key of the schema is actually valid for the request. Check the other schema locater implementations on how to do that.

Basically, use SchemaMatchType to compare the requested key against what you have and then decide to return the schema or not.

We should cover this with tests, too.

Reply:
Added check to compare the requested key using `expectedSchemaKey.Matches(key, matchType)`. I also added tests in SchemaStringLocaterTests.cpp file.